packages/cosmos-bin/src/cw3.rs

```diff
@@ -0,0 +1,434 @@
+
+    // Now create the CW3 using this CW4 as its backing group
+    let cw3 = cw3
+        .instantiate(
+            &wallet,
+            label,
+            vec![],
+            cw3_flex_multisig::msg::InstantiateMsg {
+                group_addr: cw4.get_address_string(),
+                threshold: Threshold::AbsolutePercentage {
+                    percentage: weight_needed,
+                },
+                max_voting_period: duration.0,
+                executor: None,
+                proposal_deposit: None,
+            },
+            ContractAdmin::Sender,
+        )
+        .await?;
+    log::info!("Created new CW3-flex contract: {cw3}");
+
+    // Fix permissions
+    log::info!("Fixing permissions on the contracts to make the CW3 the admin");
+    let mut builder = TxBuilder::default();
+    builder.add_update_contract_admin_mut(&cw3, &wallet, &cw3);
+    builder.add_update_contract_admin_mut(&cw4, &wallet, &cw3);
+    builder.add_execute_message_mut(
+        &cw4,
+        &wallet,
+        vec![],
+        cw4_group::msg::ExecuteMsg::UpdateAdmin {
+            admin: Some(cw3.get_address_string()),
+        },
+    )?;
+    let res = builder.sign_and_broadcast(&cosmos, &wallet).await?;
+    log::info!("Admin permissions updated in {}", res.txhash);
+
+    Ok(())
+}
+
+impl FromStr for MyDuration {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
+        let multiplier = match s.as_bytes().last().context("Duration cannot be empty")? {
+            b's' => 1,
+            b'm' => 60,
+            b'h' => 60 * 60,
+            b'd' => 60 * 60 * 24,
+            _ => anyhow::bail!("Final character in duration must be s, m, h, or d."),
+        };
+        let s = &s[0..s.len() - 1];
+        let num: u64 = s
+            .parse()
+            .with_context(|| format!("Could not parse duration value {s}"))?;
+        Ok(MyDuration(cw_utils::Duration::Time(num * multiplier)))
+    }
+}
+
+#[derive(clap::Parser)]
+struct AddMemberMessageOpt {
+    /// Members to add
+    #[clap(long)]
+    add: Vec<Address>,
+    /// Members to remove
+    #[clap(long)]
+    remove: Vec<Address>,
+    /// CW4 group contract address
+    #[clap(long)]
+    group: Address,
+}
+
+async fn update_members_message(
+    AddMemberMessageOpt { add, remove, group }: AddMemberMessageOpt,
+) -> Result<()> {
+    let msg = cw4_group::msg::ExecuteMsg::UpdateMembers {
+        add: add
+            .into_iter()
+            .map(|x| Member {
+                addr: x.get_address_string(),
+                weight: 1,
+            })
+            .collect(),
+        remove: remove.into_iter().map(|x| x.get_address_string()).collect(),
+    };
+    let msg = CosmosMsg::<Empty>::Wasm(WasmMsg::Execute {
+        contract_addr: group.get_address_string(),
+        msg: to_binary(&msg)?,
+        funds: vec![],
+    });
+    println!("{}", serde_json::to_string(&msg)?);
```

Comment:
debugging print?

Reply:
No, that's the main point of this method: printing out the message so that it can be pasted into an interface that accepts CW3 messages. I have to use a web interface for these because I need Ledger support.